### DISCUSSION TITLE
Contributing to pimcore : installation

### DISCUSSION THREAD

**Discussion by SamyMP** [pimcore/pimcore#13278] [Q&A / Support] [answered]
Hi,

I'm having trouble setting up a project to contribute to pimcore. I would like to contribute to pimcore usong the demo project, here's what I've done so far :

- Installed the demo following the docker instructions [here](https://github.com/pimcore/demo#docker)
- Forked and cloned the "pimcore/picmore" project on the "dev" folder of the demo
- Renamed the forked composer.json name to something else (ex: samymp/pimcore)
- Required my package with the "@dev" flag (essentially followed what's in [this article](https://johnbraun.blog/posts/contributing-to-a-PHP-package)

However, I keep getting errors in composer related to versions of packages like doctrine, or saying that everything is declared twice in the project (because there is pimcore/pimcore and samymp/pimcore, both declaring the same functions, ect...)

Could you please provide me a clear explanation of how to contribute to pimcore, in regards of these errors ? I'd be glad to contribute 🙂

@brusch If you could send me some guidance 🙂

**Comment by kingjia90** [comment, marked as answer]
Not sure, but It sounds like the symlink didn't correctly worked due the naming of the fork, so you might have `pimcore/pimcore` installed in the root vendor folder and `samymp/pimcore` as well symlinked as a dev package.

**Reply by SamyMP** [reply]
How can I resolve this ? I still struggle getting the project up and running with my forked version. I tried a lot of things, none of them worked, I still get conflicts about my fork being in 11.x and the dependencies being 10.5... I'll try forking the 10.5 branch then

**Reply by kingjia90** [reply]
I would make sure if you have installed your fork in the path `dev/pimcore/pimcore` and not `dev/samymp/pimcore` and `rm -rf vendor/pimcore/pimcore` to remove any composer installed package.


Then if it installs properly, there's a manual modification to do, renaming: 
https://github.com/pimcore/pimcore/blob/11.x/bundles/CoreBundle/config/routing.yml and routing_dev.yml to have file format as `.yaml`, if you are on 11.x-dev branch (at the moment there's WIP of stuff that is merged/to be merged in https://github.com/pimcore/pimcore/pull/13077 that causes problem to the demo).

**Reply by kingjia90** [reply]
In composer, the pimcore requirement must stay `"pimcore/pimcore": "11.x-dev as 10.99.99",` as the bundles are not fully supporting 11.x yet and may throw errors.

**Reply by SamyMP** [reply]
Here's what I have in my composer.json. Notice the tree of my project on the left
![image](https://user-images.githubusercontent.com/35225430/193232268-1319d2c4-4b5a-4a47-9573-bd19c42db2e5.png)

I removed `vendor/pimcore/pimcore` and ran composer update. It gave me the output on the console below on the screenshot. Still getting conflicts about symfony among other packages

Also, the `dev/pimcore/pimcore/composer.json` has `name: "samymp/pimcore"`

**Reply by kingjia90** [reply]
under `require` it should likely stay `pimcore/pimcore` on line 22

**Reply by SamyMP** [reply]
Then how do I say it needs to look in the `dev/pimcore/pimcore` to load the forked pimcore ?

**Reply by kingjia90** [reply]
the square from 53-61 is the part that decides to symlink `/dev/pimcore/pimcore` and will treat it as if it was in the `vendor` folder, but you have to revert `url` field as `"url": "./dev/*/*"`, it doesn't need to be changed. For example, we rely on this to symlink all the dev bundles as well.

if it have to use the forked or other repo, it is decided within the `dev/pimcore/pimcore` and git side, there's nothing to change on that in the composer side, at least as far as i know.

**Reply by SamyMP** [reply]
Thank you very much for your explanations @kingjia90 , the symlink seems to be working 😊

Now I have the error you were talking about with the routing files. I turned every files to `.yaml` but it doesn't seem to resolve the problem
![image](https://user-images.githubusercontent.com/35225430/193243567-26e678ee-626d-436b-a930-19d747f55b19.png)

I'd say it can't find the `@PimcoreCoreBundle` part...

**Reply by SamyMP** [reply]
I changed the routing path by putting the relative path `../../dev/pimcore/pimcore...` and it works now. Thanks again !
